### ISSUE TITLE
CWL: mergenested broken

### ISSUE THREAD

**Issue by mr-c** [DataBiosphere/toil#2846]
As reported by @kaushik-work  (thanks!)

```
class: Workflow
cwlVersion: v1.0
inputs:
  in1: string
  in2: string

steps:

  step1:
    in:
      in1: [in1, in2]
    run: list-in.cwl
    out: [out1]

outputs:
  out1: 
    type: string[]
    outputSource:
        - step1/out1
        - in1

requirements: 
  MultipleInputFeatureRequirement: {}
```

```
class: CommandLineTool
cwlVersion: v1.0
inputs:
  in1: string[]

baseCommand: [echo]

outputs:
  out1:
    type: string
    outputBinding:
      outputEval: a_$(inputs.in1)_b
```

Works with plain `cwltool` but not `toil-cwl-runner`

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-447)
┆Issue Number: TOIL-447

**Comment by adamnovak** [comment]
@DailyDreaming Didn't someone fix mergenested in a recent PR that I can't find?

**Comment by kaushik-work** [comment]
@adamnovak I can confirm that current toil master (fcce48fe8b2bef8f05865d2db586b1c8763c6f47) executes this correctly. This issue was fixed in the refactor (#2845)
